### PR TITLE
Fix G++8 warnings: catch by reference and unused variables

### PR DESCRIPTION
**Pull request by pavelkryukov**

### CODE REVIEW

**Comment by anadon** [comment]
I'll add this to the second batch of merges.  The first one is turning out rather large while I fix up build errors and I absolutely do not want anything larger.

**Comment by pavelkryukov** [comment]
These changes _are_ the build fixes.

**Comment by pavelkryukov** [comment]
GCC 8 build is blocked by https://github.com/boostorg/serialization/pull/84

**Comment by anadon** [comment]
Can you please rebase your PR on the current devel branch?  It should clean up your build errors.

**Comment by pavelkryukov** [comment]
Done.
I added a `/* weight_type w = */` comment since `w` is used in other comments.

File: include/boost/graph/maximum_adjacency_search.hpp [hunk 1]
```diff
@@ -149,7 +149,7 @@ namespace boost {
       //weight_type w;
       while (!pq.empty()) { // while PQ \neq {} do
         const vertex_descriptor u = pq.top(); // u = extractmax(PQ)
-        get(keys, u);                        vis.start_vertex(u, g);
+        /* weight_type w = */ get(keys, u);      vis.start_vertex(u, g);
```
Comment (anadon) [inline include/boost/graph/maximum_adjacency_search.hpp]: Should this comment still be here like this?

Comment (pavelkryukov) [reply include/boost/graph/maximum_adjacency_search.hpp]: See my comment above.

**Comment by jzmaddock** [comment]
This all looks good to go to me - anyone else?

**Comment by anadon** [comment]
It looks OK to me.  Still need to brush up on some of the stdlib use here, but seems good.

**Pull request merged**